cpp/src/Ice/Network.cpp

```diff
@@ -811,9 +811,16 @@ IceInternal::closeSocket(SOCKET fd)
 string
 IceInternal::addrToString(const Address& addr)
 {
-    ostringstream s;
-    s << inetAddrToString(addr) << ':' << getPort(addr);
-    return s.str();
+    if (isAddressValid(addr))
+    {
+        ostringstream s;
+        s << inetAddrToString(addr) << ':' << getPort(addr);
+        return s.str();
+    }
+    else
+    {
+        return "";
```

Comment:
This allows me to call `addrToString(addr)` blindly when throwing exceptions. If it's invalid, it's like I didn't provide an address.